doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -7,14 +7,62 @@ Security
    :local:
    :depth: 3
 
-Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into four major categories:
 
+* Trusted Execution Environment
 * Cryptography
 * Trusted storage
 * Securing production devices
 
 In the following sections you will learn more details about each listed category.
 
+Trusted Execution Environment
+*****************************
+
+In Matter samples, you can use two types of Trusted Execution Environments (TEE) to provide a secure environment for the execution of security-sensitive code, depending on the SoC:
+
+* nRF54L Series devices can use Trusted Firmware-M (TF-M).
+* nRF54H Series devices can use the Secure Domain.
```

Comment:
Cross-links missing.

Reply:
I guess there is no cross-link to the Secure Domain..

Reply:
https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_cpu.html#secure_domain